src/types.ts

```diff
@@ -87,16 +87,19 @@ export type ActiveReader = PDFActiveReader | HTMLActiveReader;
 export type ReaderReturn = InactiveReader | LoadingReader | ActiveReader;
 
 // should fetch and decrypt a resource
-export type GetContent = (href: string) => Promise<string>;
+export type GetContent<T extends string | Uint8Array> = (
```

Comment:
Let me know if there is a better way to update the type definitions that I may have overlooked.

Reply:
I think this is okay for now. I think the only other option would be splitting out separate types/arguments/returns for the `HtmlReader` and `PdfReader`, which is probably more work than it's worth right now